src/uct/tcp/tcp_ep.c

```diff
@@ -137,14 +135,21 @@ ucs_status_t uct_tcp_ep_create_connected(const uct_ep_params_t *params,
 
 void uct_tcp_ep_mod_events(uct_tcp_ep_t *ep, uint32_t add, uint32_t remove)
 {
+    int old_events = ep->events;
     int new_events = (ep->events | add) & ~remove;
 
     if (new_events != ep->events) {
         ep->events = new_events;
         ucs_trace("tcp_ep %p: set events to %c%c", ep,
                   (new_events & EPOLLIN)  ? 'i' : '-',
                   (new_events & EPOLLOUT) ? 'o' : '-');
-        uct_tcp_ep_epoll_ctl(ep, EPOLL_CTL_MOD);
+        if (new_events == 0) {
```

Comment:
why is it better to add/remove FD, than modifying his events?
AFAIK epoll is scalable in the number of FDs

Reply:
Yes, there is no difference how much fds in epoll from user space perspective.
I think from kernel perspective it would be better to have as дуыы as possible to optimize search engine (RB tree) to search fds inside the `epoll_ctl`.
Also this fixes undefined behavior when socket fd receives a disconnection event during `recv()` call - it just removes `EPOLLIN` event and closes sockets (https://github.com/openucx/ucx/blob/59a6f045ae39f093e5b6a0bb625a261d12b09ba6/src/uct/tcp/tcp_ep.c#L217)

what do you think?
Btw, we can just remove fd from epoll (inside `uct_tcp_ep_destroy`) if we detected the disconnection event.

Reply:
ok, perhaps handling fd events is independent of number of fds, but modifying and event may be..